Build failure: wineWowPackages.yabridge

### Nixpkgs version

- Unstable (26.05)

### Steps to reproduce

- Include ` yabridge`  and `yabridgectl` in `home.packages` list option of Home Manager (with `useGlobalPkgs` enabled)
- Rebuild the system
- `nixos-rebuild` complains about `/nix/store/sl0pcmal9gfxp2zbpz7n3j1ayp90lxcf-wine-wow-yabridge-9.21.drv` not being able to be built

### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

https://hydra.nixos.org/build/317963975

### Relevant log output

```console
make: *** [Makefile:329809: programs/winhlp32/x86_64-windows/hlpfile.o] Error 1
make: *** Waiting for unfinished jobs....
In file included from [01m[K/build/wine-9.21/programs/winhlp32/winhelp.h:40[m[K,
                 from [01m[K/build/wine-9.21/programs/winhlp32/winhelp.c:36[m[K:
[01m[K/build/wine-9.21/programs/winhlp32/macro.h:29:17:[m[K [01;31m[Kerror: [m[Ktwo or more data types in declaration specifiers
   29 |   BOOL          [01;31m[Kbool[m[K;
      |                 [01;31m[K^~~~[m[K
[01m[K/build/wine-9.21/programs/winhlp32/macro.h:29:21:[m[K [01;35m[Kwarning: [m[Kdeclaration does not declare anything
   29 |   BOOL          bool[01;35m[K;[m[K
      |                     [01;35m[K^[m[K
make: *** [Makefile:329918: programs/winhlp32/x86_64-windows/winhelp.o] Error 1
```

### Additional context

Probably relevant to #475479

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.18.2-zen2, NixOS, 26.05 (Yarara), 26.05.20251228.c0b0e0f`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.31.2+1`
 - nixpkgs: `/nix/store/xzl5dhddgjxp2p18h5qpkhvlkp8128j1-source`


### Notify maintainers

@avnik @bendlas @jmc-figueira @kira-bruneau @7c6f434c @reckenrode

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/applications/emulators/wine/base.nix
+++ b/pkgs/applications/emulators/wine/base.nix
@@ -221,6 +221,7 @@ stdenv.mkDerivation (
         )
       )
     );
+    env.NIX_CFLAGS_COMPILE = lib.optionalString (wineRelease == "yabridge") "-std=gnu17";
 
     # Don't shrink the ELF RPATHs in order to keep the extra RPATH
     # elements specified above.

